src/mlpack/tests/load_save_test.cpp

```diff
@@ -2527,3 +2527,132 @@ TEST_CASE("LoadCSVHeaderTest", "[LoadSaveTest]")
   REQUIRE(headers.at(2) == "c");
   REQUIRE(headers.at(3) == "d");
 }
+
+TEST_CASE("LoadCSVMissingNanTest", "[LoadSaveTest]")
+{
+  fstream f;
+  f.open("test.csv", fstream::out);
+  f << "1, , 3, 4" << std::endl;
+  f << "5, 6, 7, 8" << std::endl;
+  f << "9, 10, 11, 12" << std::endl;
+
+  arma::mat dataset;
+  data::TextOptions opts;
+  opts.Fatal() = false;
+  opts.NoTranspose() = true;
+  opts.MissingToNan() = true;
```

Comment:
Now the thing is that because this requires Armadillo 12 or newer, the test will not work on older Armadillo because the empty elements will load as 0.  So maybe the whole test should be wrapped in an `#if ARMA_VERSION_MAJOR >= 12`.  (This would be true for any test where `MissingToNan()` is true.)